crates/web-sys-async-io/src/lib.rs

```diff
@@ -21,8 +21,18 @@ fn js_value_to_io_error(error: wasm_bindgen::JsValue) -> std::io::Error {
 }
 
 #[derive(Debug, Default)]
-pub enum Op {
+pub enum WriterOp {
     #[default]
     Idle,
-    Pending(JsFuture),
+    Write(JsFuture, usize),
+    Flush(JsFuture),
+    Shutdown(JsFuture),
+}
+
+#[derive(Debug, Default)]
+pub enum ReaderOp {
+    #[default]
+    Idle,
+    Read(JsFuture),
+    ReadRemaining(js_sys::Uint8Array, usize),
 }
```

Comment:
Now that they're not shared, let's also move them to the respective `mod`s and rename back, so they're `reader::Op` and `writer::Op`